core/src/main/java/nu/yona/server/analysis/entities/WeekActivity.java

```diff
@@ -104,22 +104,33 @@ public static WeekActivity createInstance(UserAnonymized userAnonymized, Goal go
 	public static WeekActivity createInstanceInactivity(UserAnonymized userAnonymized, Goal goal, ZonedDateTime startOfWeek)
 	{
 		WeekActivity result = createInstance(userAnonymized, goal, startOfWeek);
-		// add the days
-		// notice this doesn't take care of user time zone changes during the week
-		// so for consistency it is important that the batch script adding inactivity does so
-		for (int i = 0; i < 7; i++)
+		result.addInactivityDays();
+		return result;
+	}
+
+	public void addInactivityDays() 
+	{
+		// if the batch job has already run, skip
+		if(dayActivities.size() < 7)
 		{
-			ZonedDateTime startOfDay = startOfWeek.plusDays(i);
-			if (isInFuture(startOfDay, startOfWeek.getZone()))
+			// notice this doesn't take care of user time zone changes during the week
+			// so for consistency it is important that the batch script adding inactivity does so
+			for (int i = 0; i < 7; i++)
 			{
-				break;
-			}
-			if (goal.wasActiveAtInterval(startOfDay, ChronoUnit.DAYS))
-			{
-				result.addDayActivity(DayActivity.createInstanceInactivity(userAnonymized, goal, startOfDay));
+				ZonedDateTime startOfDay = getStartTime().plusDays(i);
+				if (isInFuture(startOfDay, getStartTime().getZone()))
+				{
+					break;
+				}
+				if (getGoal().wasActiveAtInterval(startOfDay, ChronoUnit.DAYS))
+				{
+					if(!dayActivities.stream().anyMatch(dayActivity -> dayActivity.getDate().getDayOfWeek().equals(startOfDay.getDayOfWeek())))
```

Comment:
No. If there was no activity at the end of the week, this needs to be added
using the algorithm above. If there was no activity at all, the same.

On Mon, May 9, 2016 at 8:55 PM, Bert Roos <notifications@github.com> wrote:

> In core/src/main/java/nu/yona/server/analysis/entities/WeekActivity.java
> <https://github.com/yonadev/yona-server/pull/167#discussion_r62552984>:
>
> >  			{
> > -				break;
> > -			}
> > -			if (goal.wasActiveAtInterval(startOfDay, ChronoUnit.DAYS))
> > -			{
> > -				result.addDayActivity(DayActivity.createInstanceInactivity(userAnonymized, goal, startOfDay));
> > +				ZonedDateTime startOfDay = getStartTime().plusDays(i);
> > +				if (isInFuture(startOfDay, getStartTime().getZone()))
> > +				{
> > +					break;
> > +				}
> > +				if (getGoal().wasActiveAtInterval(startOfDay, ChronoUnit.DAYS))
> > +				{
> > +					if(!dayActivities.stream().anyMatch(dayActivity -> dayActivity.getDate().getDayOfWeek().equals(startOfDay.getDayOfWeek())))
>
> But now you're making the DTO on the fly for inactive days, so that code
> could go away, right?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/yonadev/yona-server/pull/167/files/8e8883d52f1261b37a85eed3b20d2a79e015b739#r62552984>
>